compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -0,0 +1,293 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// #include "OpSelector.h"
+
+#include <foder/FileLoader.h>
+
+#include <luci/Importer.h>
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+#include <luci/Import/CircleReader.h>
+
+#include <luci/Profile/CircleNodeID.h>
+
+#include <arser/arser.h>
+#include <vconone/vconone.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+void print_version(void)
+{
+  std::cout << "circle-opselector version " << vconone::get_string() << std::endl;
+  std::cout << vconone::get_copyright() << std::endl;
+}
+
+bool check_input(const std::string &str)
+{
+  bool has_hyphen = false;
+
+  if (str.empty())
+    return false;
+  if (str.at(0) == '-' || str[str.size() - 1] == '-')
+  {
+    std::cerr << "ERROR: Invalid input. Please make sure - is between the numbers" << std::endl;
+    return false;
+  }
+
+  for (char c : str)
+  {
+    if (isdigit(c))
+      continue;
+    else if (has_hyphen && c == '-') // when user enter '-' more than 2.
+    {
+      std::cerr << "ERROR: Too many '-' in str." << std::endl;
+      return false;
+    }
+    else if (c == '-')
+      has_hyphen = true;
+    else // when user enter not allowed character, print alert msg.
+    {
+      std::cerr << "ERROR: To select operator by id, please use these args: [0-9], '-', ','"
+                << std::endl;
+      return false;
+    }
+  }
+  return true;
+}
+
+void split_id_input(const std::string &str, std::vector<int> &by_id)
+{
+  std::istringstream ss;
+  ss.str(str);
+  std::string str_buf;
+
+  while (getline(ss, str_buf, ','))
+  {
+    if (check_input(str_buf)) // input validation
+    {
+      try
+      {
+        if (str_buf.find('-') == std::string::npos) // if token has no '-'
+          by_id.push_back(stoi(str_buf));
+        else // tokenize again by '-'
+        {
+          std::istringstream ss2(str_buf);
+          std::string token;
+          int from_to[2], top = 0;
+
+          while (getline(ss2, token, '-'))
+            from_to[top++] = stoi(token);
+
+          for (int number = from_to[0]; number <= from_to[1]; number++)
+            by_id.push_back(number);
+        }
+      }
+      catch (std::invalid_argument &error)
+      {
+        std::cerr << "ERROR: Invalid argument. Please make sure your input is number." << std::endl;
+        exit(EXIT_FAILURE);
+      }
+      catch (std::out_of_range)
+      {
+        std::cerr << "ERROR: Argument is out of range." << std::endl;
+        exit(EXIT_FAILURE);
+      }
+      catch (...)
```

Comment:
Could you share where these are thrown?

Reply:
Input `{non integer string}`, stoi throw upper error. But non integer is already handled in `check_input`. I will delete.
Input `{too big integer}` throw lower error.